### PR TITLE
Fix bug in cancel function

### LINKED ISSUE #17843: Noninteger coefficients in cancel function
Documentation says that in output of 'cancel' function "leading coefficients of numerator and denominator are integers" which isn't always true. For example
`>>> cancel('(x/2+1)/y')`
returns
`(x + 2)/(2*y)` 
which is fine, but
`>>> cancel('(x/2+1)')`
returns
`x/2 + 1`
and I would expect 
`(x+2)/2`

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

Fixes #17843

#### Brief description of what is fixed or changed
Fix bug in cancel function

#### Release Notes

*  simplify
   *  Fixing bug in cancel function

### CODE REVIEW

**Comment by namannimmo10** [comment]
@oscarbenjamin The tests are failing in some other files involving cancel function. 
https://travis-ci.org/sympy/sympy/jobs/628826029?utm_medium=notification&utm_source=github_status

Like in https://github.com/sympy/sympy/blob/0bffa281e62b4d29fbe3cd22faa4d612a4b1ca76/sympy/simplify/ratsimp.py#L20
`(x + y)/(x*y)` is expected but the actual result is: `(1/(x*y))*(x + y)`

**Comment by oscarbenjamin** [comment]
Maybe what is needed is something that checks specifically for the case of a 2-arg Mul with a Rational argument...

**Comment by namannimmo10** [comment]
Initially, the `cancel` function showed the wrong output if we passed the argument which was not in `p / q` form.
The change we made fixes that problem but now shows the wrong output for the argument which is in `p / q` form. 

Maybe we could check in the beginning whether the argument is in `p / q` form or not. And then return accordingly.

File: sympy/polys/polytools.py [hunk 1]
```diff
@@ -6651,7 +6651,7 @@ def cancel(f, *gens, **args):
     c, P, Q = F.cancel(G)
 
     if not isinstance(f, (tuple, Tuple)):
-        return c*(P.as_expr()/Q.as_expr())
+        return c * Mul(1/Q.as_expr(), P.as_expr(), evaluate=False)
```
Comment (smichr) [inline sympy/polys/polytools.py]:
```suggestion
        return Mul(c, P.as_expr()/Q.as_expr(), evaluate=False)
```

Comment (namannimmo10) [reply sympy/polys/polytools.py]: I tried, didn't work out

Comment (namannimmo10) [reply sympy/polys/polytools.py]: https://github.com/sympy/sympy/issues/17843#issuecomment-568475591

Comment (smichr) [reply sympy/polys/polytools.py]: The one test that failed will just require modifying the expected result to be `Mul((-S.Half, x + sqrt(2)*x - 2), evaluate=False)`

**Comment by smichr** [comment]
I ventured into modifications because you said there was 1 failure...it must have been only in that file because there were lots of failures elsewhere. Let's see how the current mod turns out.

**Comment by namannimmo10** [comment]
> I ventured into modifications because you said there was 1 failure...

That was before I raised this PR. It was a part of the discussion [here](https://github.com/sympy/sympy/issues/17843#issuecomment-568475591).  There were so many tests failures in other files, as mentioned [above](https://github.com/sympy/sympy/pull/18107#issuecomment-569104871).

I think the problem here is that a 2-arg Mul distributes Rationals.

**Comment by smichr** [comment]
If you end up fixing these tests, we are a step closer to removing autodistribution of the Rational in a 2-arg Mul...but I'm not sure it's worth it.

**Comment by namannimmo10** [comment]
What do you think about [this](https://github.com/sympy/sympy/pull/18107#issuecomment-569135667) or [this](https://github.com/sympy/sympy/issues/17843#issuecomment-568450931) one??

**Comment by smichr** [comment]
> What do you think about [this](https://github.com/sympy/sympy/pull/18107#issuecomment-569135667) or [this](https://github.com/sympy/sympy/issues/17843#issuecomment-568450931) one??

I think it is easier to write a caveat in the docstring:

```
Note: due to automatic distribution of Rationals, a sum divided by an integer will appear as a sum. To recover a rational form use `together` on the result:

>>> cancel(x/2 + 1)
x/2 + 1
>>> together(_)
(x + 1)/2
```

When you look at the failures you have to ask whether the rational form in most cases is preferred over the distributed form. When I have looked, I usually prefer the distributed form. And when I care, I use `together`.

**Comment by namannimmo10** [comment]
Sorry, I couldn't fix the failing tests.  But this looks good right now. Thanks for trying with me.

**Comment by namannimmo10** [comment]
Please merge this, @smichr.

**Comment by smichr** [comment]
If anyone thinks it worthwhile, a keyword could be used to make sure the rational form is returned (and it would default to *not* doing so to keep with current expectations. e.g. `cancel(x/2 +1, strict=True)`. So let's see if there are any thoughts about this from anyone else.

**Comment by smichr** [comment]
I guess this can go in now...it is easy to modify the docstring if a decision to add a keyword is made at some time.

**Comment by namannimmo10** [comment]
OK.  Please include me in the discussion whenever that happens.

**Pull request merged**